chemprop/models/mpn.py

```diff
@@ -169,9 +161,25 @@ def forward(self,
         :param features_batch: A list of numpy arrays containing additional features.
         :return: A PyTorch tensor of shape :code:`(num_molecules, hidden_size)` containing the encoding of each molecule.
         """
-        if type(batch) != BatchMolGraph:
-            batch = mol2graph(batch)
+        if type(batch[0]) != BatchMolGraph:
+            batch = [mol2graph(b) for b in batch]
+
+        if self.use_input_features:
+            features_batch = torch.from_numpy(np.stack(features_batch)).float().to(self.device)
+
+            if self.features_only:
+                return features_batch
 
-        output = self.encoder.forward(batch, features_batch)
+        output = None
+        for enc, ba in zip(self.encoder, batch):
+            _output = enc.forward(ba)
+            output = _output if output is None else torch.cat([output, _output], dim=1)
+
+        if self.use_input_features:
+            if len(features_batch.shape) == 1:
+                features_batch = features_batch.view([1, features_batch.shape[0]])
+            # todo not sure if this did anything
+            # features_batch = features_batch.to(mol_vecs)
```

Comment:
You can leave this out since `features_batch` is already moved to the right device on line 168.